### PR TITLE
[Feature Fix] Update to view link icons

### PR DESCRIPTION
**Pull request by leb2dg**
# Purpose

This resolves issue #2971. The icons that are displayed on the sharing page under View-only Links are outdated.

# Changes

This requires the custom binding called `getIcon` defined in *osfHelpers.js* from PR #2953. The category of the node is added to what is returned by `to_json` in *model.py*. This is necessary so that the category can be given to `getIcon` in *contributors.mako*. The old image code has been taken out of *model.py*.

# Side effects

Under View-only Links on the project sharing page, the updated icon is displayed next to the title of the component. There is an added field to what is returned by `to_json`.

Fix:
screen shot 2015-06-08 at 11 20 34 am

### CODE REVIEW

File: website/project/model.py [hunk 1]
```diff
@@ -2640,7 +2640,7 @@ def to_json(self):
             "key": self.key,
             "name": self.name,
             "creator": {'fullname': self.creator.fullname, 'url': self.creator.profile_url},
-            "nodes": [{'title': x.title, 'url': x.url, 'scale': str(self.node_scale(x)) + 'px', 'imgUrl': self.node_icon(x)}
+            "nodes": [{'title': x.title, 'url': x.url, 'scale': str(self.node_scale(x)) + 'px', 'imgUrl': self.node_icon(x), 'category': x.category}
```
Comment (samanehsan) [inline website/project/model.py]: Can `'imgUrl': self.node_icon(x)` be removed here since it is no longer being used in `contributors.mako`?

**Comment by samanehsan** [comment]
Done with first pass! :palm_tree:

File: website/project/model.py (hunk 1 above)
Comment (leb2dg) [inline website/project/model.py]: When I removed that the formatting was messed up but I can look into why (possibly a CSS issue).

**Comment by leb2dg** [comment]
I was confused...I had removed the scale last time and that was what caused the format change. The unused function has now been removed!

**Comment by samanehsan** [comment]
Thanks! @sloria, this is ready for another pass! :sunny:

**Pull request merged**